test/integration/framework/plant_operations.go

```diff
@@ -0,0 +1,268 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	"k8s.io/client-go/util/retry"
+
+	"k8s.io/api/core/v1"
+
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/sirupsen/logrus"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+)
+
+// NewPlantTest creates a new plantGardenerTest object, given an already created plant (created after parsing a plant YAML) and a path to a kubeconfig of an external cluster
+func NewPlantTest(kubeconfig string, kubeconfigPathExternalCluster string, plant *gardencorev1alpha1.Plant, logger *logrus.Logger) (*PlantTest, error) {
+	if len(kubeconfig) == 0 {
+		return nil, fmt.Errorf("Please specify the kubeconfig path correctly")
+	}
+
+	if len(kubeconfigPathExternalCluster) == 0 {
+		return nil, fmt.Errorf("Please specify the kubeconfig path for the external cluster correctly")
+	}
+
+	k8sGardenClient, err := kubernetes.NewClientFromFile("", kubeconfig, client.Options{
+		Scheme: kubernetes.GardenScheme,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &PlantTest{
+		GardenClient:                  k8sGardenClient,
+		Plant:                         plant,
+		kubeconfigPathExternalCluster: kubeconfigPathExternalCluster,
+		Logger:                        logger,
+	}, nil
+}
+
+// CreatePlantSecret creates a new Secret for the Plant
+func (s *PlantTest) CreatePlantSecret(ctx context.Context) error {
+
+	if len(s.kubeconfigPathExternalCluster) == 0 {
+		return fmt.Errorf("Path to kubeconfig of external cluster not set")
+	}
+
+	kubeconfigExternalCluster, err := ioutil.ReadFile(s.kubeconfigPathExternalCluster)
+	if err != nil {
+		s.Logger.Errorf("Unable to read kubeconfig for external cluster from '%s'", s.kubeconfigPathExternalCluster)
+		return err
+	}
+
+	plantSecretName := "test-secret-plant-" + s.Plant.Name
+	plantSecret := v1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: s.Plant.Namespace, Name: plantSecretName}}
+	plantSecret.Data = make(map[string][]byte)
+	plantSecret.Data["kubeconfig"] = kubeconfigExternalCluster
+
+	err = s.GardenClient.Client().Create(ctx, &plantSecret)
+	if err != nil {
+		if !apierrors.IsAlreadyExists(err) {
+			s.Logger.Errorf("Unable to create the plant secret %+v: %v", plantSecret, err)
+			return err
+		}
+		err := s.GardenClient.Client().Get(ctx, client.ObjectKey{Namespace: plantSecret.Namespace, Name: plantSecret.Name}, &plantSecret)
+		if err != nil {
+			return err
+		}
+	}
+	s.PlantSecret = &plantSecret
+	return nil
+}
+
+// DeletePlantSecret deletes the Secret of the Plant
+func (s *PlantTest) DeletePlantSecret(ctx context.Context) error {
+	err := s.GardenClient.Client().Delete(ctx, s.PlantSecret)
```

Comment:
Why? This method does not add that much value.
Also, just directly return the `err` instead of the check.